tests/scroll.test.ts

```diff
@@ -0,0 +1,58 @@
+import { Browser, Page } from 'playwright';
+import { launch } from '../src/launch';
+import { getScrollValue, scroll } from '../src/scroll';
+import { TEST_URL } from './utils';
+
+describe('scroll', () => {
+  let browser: Browser;
+  let page: Page;
+
+  beforeAll(async () => {
+    browser = await launch();
+    page = await browser.newPage();
+  });
+
+  afterAll(() => browser.close());
+
+  it('scrolls element to specified position if possible', async () => {
+    await page.goto(`${TEST_URL}infinite-scroll`);
+
+    await scroll(page, 'html', { x: 0, y: 200 });
+
+    const elementHandle = await page.waitForSelector('html');
+    expect(await getScrollValue(page, elementHandle)).toEqual({ x: 0, y: 200 });
+  });
+
+  it('handles infinite scroll', async () => {
+    await page.goto(`${TEST_URL}infinite-scroll`);
+
+    await scroll(page, 'html', { x: 0, y: 2000 });
+
+    const elementHandle = await page.waitForSelector('html');
+    expect(await getScrollValue(page, elementHandle)).toEqual({
+      x: 0,
+      y: 2000,
+    });
+  });
+
+  it('does not throw error if can partially scroll element', async () => {
+    await page.goto(`${TEST_URL}large`);
+
+    await scroll(page, 'html', { x: 0, y: 2000, timeout: 1000 });
```

Comment:
Can you add a test where you navigate while it is trying to scroll to make sure it does not throw an error. Like this test, but do not await the scroll promise, and do a goto.

Reply:
ok

Reply:
Nevermind lets throw the error for now

Reply:
Then I think the way to fix it is to add this first, and take the waitUntil options and pass them through (defaulting to load the same)

```
await page.waitForLoadState
```